ibis/backends/trino/datatypes.py

```diff
@@ -84,99 +72,30 @@ def _floating(element, compiler, **kw):
-                .skip(RPAREN)
-                .optional(default_temporal_scale),
-                timezone=spaceless_string("with time zone").result("UTC").optional(),
-            ).optional(dict(scale=default_temporal_scale, timezone=None))
-        )
-        .combine_dict(partial(dt.Timestamp))
-    )
-
-    primitive = (
-        spaceless_string("interval year to month").result(dt.Interval(unit="M"))
-        | spaceless_string("interval day to second").result(dt.Interval(unit="ms"))
-        | spaceless_string("bigint").result(dt.int64)
-        | spaceless_string("boolean").result(dt.boolean)
-        | spaceless_string("varbinary").result(dt.binary)
-        | spaceless_string("double").result(dt.float64)
-        | spaceless_string("real").result(dt.float32)
-        | spaceless_string("smallint").result(dt.int16)
-        | spaceless_string("date").result(dt.date)
-        | spaceless_string("tinyint").result(dt.int8)
-        | spaceless_string("integer").result(dt.int32)
-        | spaceless_string("uuid").result(dt.uuid)
-        | spaceless_string("json").result(dt.json)
-        | spaceless_string("ipaddress").result(dt.inet)
-    )
-
-    varchar = (
-        spaceless_string("varchar", "char")
-        .then(LPAREN.then(LENGTH).skip(RPAREN).optional())
-        .result(dt.string)
-    )
-
-    decimal = spaceless_string("decimal", "numeric").then(
-        parsy.seq(LPAREN.then(PRECISION).skip(COMMA), SCALE.skip(RPAREN))
-        .optional(default_decimal_parameters)
-        .combine(dt.Decimal)
-    )
-
-    time = (
-        spaceless_string("time").then(
-            parsy.seq(
-                scale=LPAREN.then(TEMPORAL_SCALE)
-                .skip(RPAREN)
-                .optional(default_temporal_scale),
-                timezone=spaceless_string("with time zone").result("UTC").optional(),
-            ).optional(dict(scale=default_temporal_scale, timezone=None))
-        )
-        # TODO: support time with precision
-        .result(dt.time)
-    )
-
-    ty = parsy.forward_declaration()
-
-    array = spaceless_string("array").then(LPAREN).then(ty).skip(RPAREN).map(dt.Array)
-    map = spaceless_string("map").then(
-        parsy.seq(LPAREN.then(ty).skip(COMMA), ty.skip(RPAREN)).combine(dt.Map)
-    )
-
-    struct = (
-        spaceless_string("row")
-        .then(LPAREN)
-        .then(parsy.seq(spaceless(FIELD), ty).sep_by(COMMA).map(dt.Struct.from_tuples))
-        .skip(RPAREN)
-    )
+class TrinoTypeParser(TypeParser, Final):
+    dialect = "trino"
 
-    ty.become(primitive | timestamp | time | varchar | decimal | array | map | struct)
-    return ty.parse(text)
+    default_decimal_precision = 18
+    default_decimal_scale = 3
+    default_temporal_scale = 3
 
 
-_from_trino_types = {
-    DOUBLE: dt.Float64,
-    sat.REAL: dt.Float32,
-    JSON: dt.JSON,
-}
+def parse(text: str) -> dt.DataType:
+    """Parse a Trino type into an ibis data type."""
+    return TrinoTypeParser().parse(text)
```

Comment:
We wouldn't need the parse function if we could use `TrinoTypeParser.parse()` as a classmethod directly, see my comments at `TypeParser`'s definition.

Reply:
We need to keep the function, because its the uniform interface to parsing a backends types, and not all backends need to or can use `TypeParser`.

Reply:
Sure, this is similar to the `_to_sqla_types` and `_from_sqla_types` classmethods in the base translator classes.